python/wakapi_anyide/watchers/filewatcher.py

```diff
@@ -72,82 +77,87 @@ async def _task(self, queue: Queue[Event]):
-                    self.cache[resolved_path] = File(
-                        resolved_path,
-                        b"",
-                        0
-                    )
-                
-                if self.current_file is None:
-                    self.current_file = new_file
                     
-                    continue
+                    # add the file to the cache only for the diffing, making sure it is empty
+                    self.cache[resolved_path] = FileMetadata.empty(resolved_path)
+            
+                # Now, add the file current one as being handled
+                self.current_file = new_file
                 
-                if self.current_file.path != resolved_path:
-                    logger.debug(f"File changed (was {self.current_file.path}, now {resolved_path})")
-                    event = process_file_change(
-                        new_file=new_file,
-                        old_file=self.cache[resolved_path],
-                        time=time.time(),
-                        env=self.env
-                    )
-                    
-                    if event is not None:
-                        await queue.put(event)
-                    
-                    self.cache[resolved_path] = new_file
+                # For some reason, all the files to be processed are handled here, but the last file is the .current file
+                # Oh wait, that makes complete sense! 
+                
+                # Process the current file and add it to the eventqueue
+                event = process_file_change(
+                    new_file=self.current_file,
+                    old_file=self.cache[resolved_path],
+                    time=time.time(),
+                    env=self.env
+                )
+                
+                if event is not None:
+                    await queue.put(event)
                     
-                self.current_file = new_file
+                if self.delete:
+                    del self.cache[resolved_path]
+                    logger.info(f"Deleted file: {format_file(self.current_file)} ")
+                
         
     async def setup(self, tg: TaskGroup, emit_event: Queue[Event]):
         self.task = tg.create_task(self._task(emit_event))
     
     async def shutdown(self):
         pass
     
-    async def resolve_events(self) -> AsyncGenerator[Event, None]:
-        if self.current_file is not None:
-            event = process_file_change(
-                new_file=self.current_file,
-                old_file=self.cache[self.current_file.path],
-                time=time.time(),
-                env=self.env
-            )
-            
-            if event is not None:
-                yield event
-            
+    async def resolve_events(self, events: Dict[str, Event]):
+        # I think the actual goal of resolve events is to complete the last file in the queue
```

Comment:
The goal of resolve events is to allow watchers to submit events just before the heartbeats are sent.